bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/BookieShell.java

```diff
@@ -927,12 +930,26 @@ int runCmd(CommandLine cmdLine) throws Exception {
                 Iterator<Map.Entry<Long, List<String>>> iter = underreplicationManager
                         .listLedgersToRereplicate(predicate, printMissingReplica);
                 while (iter.hasNext()) {
-                    System.out.println(ledgerIdFormatter.formatLedgerId(iter.next().getKey()));
+                    Map.Entry<Long, List<String>> urLedgerMapEntry = iter.next();
+                    long urLedgerId = urLedgerMapEntry.getKey();
+                    System.out.println(ledgerIdFormatter.formatLedgerId(urLedgerId));
                     if (printMissingReplica) {
-                        iter.next().getValue().forEach((missingReplica) -> {
-                            System.out.println("\t" + missingReplica);
+                        urLedgerMapEntry.getValue().forEach((missingReplica) -> {
+                            System.out.println("\tMissingReplica : " + missingReplica);
                         });
                     }
+                    if (printReplicationWorkerId) {
+                        try {
+                            String replicationWorkerId = underreplicationManager
+                                    .getReplicationWorkerIdRereplicatingLedger(urLedgerId);
+                            if (replicationWorkerId != null) {
+                                System.out.println("\tReplicationWorkerId : " + replicationWorkerId);
```

Comment:
Would it be useful to print something even in case of missing worker

Reply:
i think it would be noisy. At a given point, the number of locks acquired would be roughly equal to number of ReplicationWorkers in the cluster. So for the rest of the under-replicated ledgers this value would be null and if we print something in case lock is not yet acquired for the under replicated ledger then it would become verbose and noisy.